docs/deployment/private/self-hosted-deployment.mdx

```diff
@@ -0,0 +1,272 @@
+
+:::info
+
+
+- `<build_version>`: This was part of the output when you ran `ddn supergraph build create` command.
+
+- `<domain>`: Domain which will be used for accessing this ingress.  This could be constructed in the following format: **https://`<build_version>`.`<your_fqdn>`**.
+
+- `<namespace>`: Namespace where your v3-engine will be deployed to.
+
+- `<v3-engine-helm-release-name>`: Helm release name for your v3-engine.
+:::
+
+```bash
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-prod
+    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+  labels:
+    app: v3-engine-<build_version>
+  name: v3-engine-<build_version>
+  namespace: <namespace>
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: <domain>
+    http:
+      paths:
+      - backend:
+          service:
+            name: <v3-engine-helm-release-name>-v3-engine
+            port:
+              number: 3000
+        path: /
+        pathType: ImplementationSpecific
+  tls:
+  - hosts:
+    - <domain>
+    secretName: <domain>-tls-certs
+```
+
+Next, you will be running the command below in order to record the ingress URL within Hasura's Control Plane.
+
+```bash title="Record the ingress URL (Make sure to prepend with protocol) for your build version within Hasura's control plane."
+ddn supergraph build set-self-hosted-engine-url <ingress-url> --build-version <build_version> --project <ddn-project-name>
+```
+
+In the following step, you will create an ingress which will serve as your project's main URL.
+
+**NOTE:** We are once again using an example of an ingress object which will work provided you have <b>nginx</b> and <b>cert-manager</b> installed on your cluster.
+
+:::info
+
+
+- `<namespace>`: Namespace where your v3-engine will be deployed to.
+
+- `<domain>`: Domain which will be used for accessing this ingress.  This could be constructed in the following format: **https://`<prod-api>`.`<your_fqdn>`**.
+
+- `<v3-engine-helm-release-name>`: Helm release name for your v3-engine.
+:::
+
+```bash
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-prod
+    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+  labels:
+    app: v3-engine
+  name: v3-engine
+  namespace: <namespace>
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: <domain>
+    http:
+      paths:
+      - backend:
+          service:
+            name: <v3-engine-helm-release-name>-v3-engine
+            port:
+              number: 3000
+        path: /
+        pathType: ImplementationSpecific
+  tls:
+  - hosts:
+    - <domain>
+    secretName: <domain>-tls-certs
+```
+
+```bash title="Record the ingress URL (Make sure you prepend with protocol) for your main project URL within Hasura's control plane."
+ddn project set-self-hosted-engine-url <ingress-url>
+```
+
+## Step 6. Deploy via Helm
```

Comment:
Should we indicate some helm commands here? I know we have it in get started for the chart, but it would be nice to have the commands here so that I can follow along.

Reply:
Yup, makes sense.  Let me work on that